src/engine/contactwriter.cpp

```diff
@@ -2530,6 +2546,64 @@ static ContactWriter::DetailList allSingularDetails()
     return details;
 }
 
+static QContactDetail supportedDetailInstance(QContactDetail::DetailType type)
+{
+    switch (type) {
```

Comment:
No, as it only returns a valid detail if the detail type is in the supported list.  eg, QContactVersion isn't supported by our backend, etc.